views/js/checkout-component-renderer.js

```diff
@@ -495,19 +495,19 @@ jQuery(document).ready(function() {
          * @param state
          */
         function handleOnChange(state) {
-            placeOrderAllowed = state.isValid;
+            placeOrderAllowed[state.data.paymentMethod.type] = state.isValid;
         }
 
         function handleOnAdditionalDetails(state) {
             hidePopup();
-            processPaymentsDetails(state.data).done(function(responseJSON) {
+            processPaymentsDetails(state.data).done(function (responseJSON) {
                 processControllerResponse(responseJSON,
                     getSelectedPaymentMethod());
             });
         }
 
-        function handleOnSubmit(state, component) {
-            placeOrderAllowed = state.isValid;
+        function handleOnSubmit(state/*, component*/) {
```

Comment:
In case it's not used we can remove it.

```suggestion
        function handleOnSubmit(state) {
```